gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -146,13 +144,12 @@ private void publishFileSet(CopyableFile.DatasetAndPartition datasetAndPartition
 
     CopyableDatasetMetadata metadata = CopyableDatasetMetadata.deserialize(
         datasetWorkUnitStates.iterator().next().getProp(CopySource.SERIALIZED_COPYABLE_DATASET));
-    Path datasetWriterOutputPath = new Path(new Path(writerOutputDir, datasetAndPartition.identifier()),
-        PathUtils.withoutLeadingSeparator(metadata.getDatasetTargetRoot()));
+    Path datasetWriterOutputPath = new Path(this.writerOutputDir, datasetAndPartition.identifier());
 
     log.info(String
         .format("Publishing fileSet from %s to %s", datasetWriterOutputPath, metadata.getDatasetTargetRoot()));
 
-    HadoopUtils.renameRecursively(fs, datasetWriterOutputPath, metadata.getDatasetTargetRoot());
+    HadoopUtils.renameRecursively(fs, datasetWriterOutputPath, findPathRoot(metadata.getDatasetTargetRoot()));
```

Comment:
Does this change the publish behavior? Shouldn't we publish to the DatasetTargeRoot path defined from source?